src/gui/plot/qgsplotwidget.h

```diff
@@ -0,0 +1,139 @@
+/***************************************************************************
+    qgsplotwidget.h
+    ---------------
+    begin                : August 2025
+    copyright            : (C) 2025 by Mathieu Pellerin
+    email                : mathieu dot opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSPLOTWIDGET_H
+#define QGSPLOTWIDGET_H
+
+#include "qgis_sip.h"
+#include "qgspanelwidget.h"
+#include "qgsplot.h"
+#include "qgsnumericformat.h"
+
+#include <QWidget>
+
+/**
+ * \ingroup gui
+ * \class QgsPlotWidget
+ * \brief Base class for widgets which allow control over the properties of plots.
+ * \since QGIS 4.0
+ */
+class GUI_EXPORT QgsPlotWidget : public QgsPanelWidget
+{
+    Q_OBJECT
+
+  public:
+    /**
+     * Constructor for QgsPlotWidget.
+     * \param parent parent widget
+     */
+    QgsPlotWidget( QWidget *parent = nullptr )
+      : QgsPanelWidget( parent )
+    {}
+
+    /**
+     * Sets the \a plot to show in the widget. Ownership is not transferred.
+     * \see plot()
+     */
+    virtual void setPlot( QgsPlot *plot ) = 0;
+
+    /**
+     * Returns the plot defined by the current settings in the widget.
```

Comment:
Add note about ownership/factory, and consider renaming to "createPlot" if it's a factory